[WIP] Investigate `mark_used(attr)`

petrochenkov opened the pull request:
r? @ghost

petrochenkov commented:
@bors try @rust-timer queue

rust-timer commented:
Awaiting bors try build completion

bors commented:
:hourglass: Trying commit 5a7416d56af30cb913f0efcfc4fe7a8c950e934c with merge e1489337a11ba38720eba65fe1dd549de224ab40...
<!-- homu: {"type":"TryBuildStarted","head_sha":"5a7416d56af30cb913f0efcfc4fe7a8c950e934c","merge_sha":"e1489337a11ba38720eba65fe1dd549de224ab40"} -->

bors commented:
:sunny: Try build successful - [checks-actions](https://github.com/rust-lang-ci/rust/runs/929193033), [checks-azure](https://dev.azure.com/rust-lang/e71b0ddf-dd27-435a-873c-e30f86eea377/_build/results?buildId=35699)
Build commit: e1489337a11ba38720eba65fe1dd549de224ab40 (`e1489337a11ba38720eba65fe1dd549de224ab40`)
<!-- homu: {"type":"TryBuildCompleted","builders":{"checks-azure":"https://dev.azure.com/rust-lang/e71b0ddf-dd27-435a-873c-e30f86eea377/_build/results?buildId=35699","checks-actions":"https://github.com/rust-lang-ci/rust/runs/929193033"},"merge_sha":"e1489337a11ba38720eba65fe1dd549de224ab40"} -->

rust-timer commented:
Queued e1489337a11ba38720eba65fe1dd549de224ab40 with parent 6b09c37ddc240e25748e53d7a8f81f252def6dee, future [comparison URL](https://perf.rust-lang.org/compare.html?start=6b09c37ddc240e25748e53d7a8f81f252def6dee&end=e1489337a11ba38720eba65fe1dd549de224ab40).

rust-timer commented:
Finished benchmarking try commit (e1489337a11ba38720eba65fe1dd549de224ab40): [comparison url](https://perf.rust-lang.org/compare.html?start=6b09c37ddc240e25748e53d7a8f81f252def6dee&end=e1489337a11ba38720eba65fe1dd549de224ab40).

Benchmarking this pull request likely means that it is perf-sensitive, so we're automatically marking it as not fit for rolling up. Please note that if the perf results are neutral, you should likely undo the rollup=never given below by specifying `rollup-` to bors.

Importantly, though, if the results of this run are non-neutral **do not** roll this PR up -- it will mask other regressions or improvements in the roll up.

@bors rollup=never

petrochenkov commented:
@bors try @rust-timer queue

rust-timer commented:
Awaiting bors try build completion

bors commented:
:hourglass: Trying commit ecf5823911045b2beff2ea829d070fcfb205b734 with merge b514752e6643f54cbcf1fc1685e0fe4c4e6e3c4d...
<!-- homu: {"type":"TryBuildStarted","head_sha":"ecf5823911045b2beff2ea829d070fcfb205b734","merge_sha":"b514752e6643f54cbcf1fc1685e0fe4c4e6e3c4d"} -->

bors commented:
:sunny: Try build successful - [checks-actions](https://github.com/rust-lang-ci/rust/runs/935192909), [checks-azure](https://dev.azure.com/rust-lang/e71b0ddf-dd27-435a-873c-e30f86eea377/_build/results?buildId=35726)
Build commit: b514752e6643f54cbcf1fc1685e0fe4c4e6e3c4d (`b514752e6643f54cbcf1fc1685e0fe4c4e6e3c4d`)
<!-- homu: {"type":"TryBuildCompleted","builders":{"checks-azure":"https://dev.azure.com/rust-lang/e71b0ddf-dd27-435a-873c-e30f86eea377/_build/results?buildId=35726","checks-actions":"https://github.com/rust-lang-ci/rust/runs/935192909"},"merge_sha":"b514752e6643f54cbcf1fc1685e0fe4c4e6e3c4d"} -->

rust-timer commented:
Queued b514752e6643f54cbcf1fc1685e0fe4c4e6e3c4d with parent b5eae9c44d713779a08e6db352088d45cad3e9b6, future [comparison URL](https://perf.rust-lang.org/compare.html?start=b5eae9c44d713779a08e6db352088d45cad3e9b6&end=b514752e6643f54cbcf1fc1685e0fe4c4e6e3c4d).

rust-timer commented:
Finished benchmarking try commit (b514752e6643f54cbcf1fc1685e0fe4c4e6e3c4d): [comparison url](https://perf.rust-lang.org/compare.html?start=b5eae9c44d713779a08e6db352088d45cad3e9b6&end=b514752e6643f54cbcf1fc1685e0fe4c4e6e3c4d).

Benchmarking this pull request likely means that it is perf-sensitive, so we're automatically marking it as not fit for rolling up. Please note that if the perf results are neutral, you should likely undo the rollup=never given below by specifying `rollup-` to bors.

Importantly, though, if the results of this run are non-neutral **do not** roll this PR up -- it will mask other regressions or improvements in the roll up.

@bors rollup=never

petrochenkov committed:
rustc_ast: `(Nested)MetaItem::check_name` -> `has_name`

For consistency with `Attribute::has_name` which doesn't mark the attribute as used either.

Replace all uses of `check_name` with `has_name` outside of rustc

petrochenkov committed:
WIP
